src/assets/wise5/services/tagService.ts

```diff
@@ -80,4 +82,25 @@ export class TagService {
   hasTagName(tagName: string): boolean {
     return this.getExistingTagNames().includes(tagName);
   }
+
+  retrieveUserTags(): Observable<any> {
+    return this.http.get(`/api/user/tags`);
+  }
+
+  applyTagToProjects(tag: any, projects: Project[]): Subscription {
+    const projectIds = projects.map((project) => project.id);
+    return this.http.put(`/api/projects/tag/${tag}`, projectIds).subscribe(() => {});
```

Comment:
No need to pass in a function?
```suggestion
    return this.http.put(`/api/projects/tag/${tag}`, projectIds).subscribe();
```